modular_skyrat/modules/clock_cult/code/structures/tinkerers_cache.dm

```diff
@@ -0,0 +1,98 @@
+/obj/structure/destructible/clockwork/gear_base/tinkerers_cache
+	name = "tinkerer's cache"
+	desc = "A bronze store filled with parts and components."
+	icon_state = "tinkerers_cache"
+	clockwork_desc = "Can be used to forge powerful Ratvarian items at the cost of power and time."
+	anchored = TRUE
+	break_message = span_warning("The tinkerer's cache melts into a pile of brass.")
+	COOLDOWN_DECLARE(use_cooldown)
+	/// Assoc list of the names of all the craftable items to their path
+	var/static/list/craft_possibilities
+
+
+/obj/structure/destructible/clockwork/gear_base/tinkerers_cache/Initialize(mapload)
+	. = ..()
+	if(!length(craft_possibilities))
+		assemble_datum_list()
+
+
+/obj/structure/destructible/clockwork/gear_base/tinkerers_cache/attack_hand(mob/living/user)
+	. = ..()
+	if(.)
+		return
+
+	if(!IS_CLOCK(user))
+		to_chat(user, span_warning("You try to put your hand into [src], but almost burn yourself!"))
+		return
+
+	if(!anchored)
+		to_chat(user, span_brass("[src] needs to be anchored to the floor first."))
+		return
+
+	if(depowered)
+		to_chat(user, span_brass("[src] isn't connected to power!"))
+		return
+
+	if(!COOLDOWN_FINISHED(src, use_cooldown))
+		to_chat(user, span_brass("[src] is still warming up, it will be ready in [DisplayTimeText(COOLDOWN_TIMELEFT(src, use_cooldown))]."))
+		return
+
+	var/selection = tgui_input_list(user, "Select an item to create at the forge.", "Forging", craft_possibilities)
+	var/datum/tinker_cache_item/chosen_item = craft_possibilities[selection]
```

Comment:
Sanitize this, make sure that selection is not null. I don't think it's actually going to runtime, but y'know.

Reply:
done